core/src/types/builder.rs

```diff
@@ -21,16 +21,25 @@ use std::env;
 use crate::raw::*;
 use crate::*;
 
-/// Builder is used to build a storage accessor used by [`Operator`].
+/// Builder is used to set up a real underlying service, i.e. storage accessor.
+///
+/// One builder is usually used by [`Operator`] during its initialization.
+/// It can be created by accepting several k-v pairs from one HashMap, one iterator and specific environment variables.
+///
+/// By default each builder of underlying service must support deriving from one HashMap.
+/// Besides that, according to the implementation, each builder may have its own special methods
```

Comment:
Every builder MUST have it's own special methods and `from_map` and other API will be built upon those methods.

Reply:
fixed. thank you so much